tests/functional/watcherdecisionengine_controller_test.go

```diff
@@ -0,0 +1,181 @@
+			WatcherDecisionEngine := GetWatcherDecisionEngine(watcherTest.WatcherDecisionEngine)
+			Expect(WatcherDecisionEngine.Status.ObservedGeneration).To(Equal(int64(0)))
+		})
+
+		It("should have a finalizer", func() {
+			// the reconciler loop adds the finalizer so we have to wait for
+			// it to run
+			Eventually(func() []string {
+				return GetWatcherDecisionEngine(watcherTest.WatcherDecisionEngine).Finalizers
+			}, timeout, interval).Should(ContainElement("openstack.org/watcherdecisionengine"))
+		})
+
+	})
+})
+
+var _ = Describe("WatcherDecisionEngine controller", func() {
+	When("A WatcherDecisionEngine instance is created", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateWatcherDecisionEngine(watcherTest.WatcherDecisionEngine, GetDefaultWatcherDecisionEngineSpec()))
+		})
+
+		It("should have the Spec fields defaulted", func() {
+			WatcherDecisionEngine := GetWatcherDecisionEngine(watcherTest.WatcherDecisionEngine)
+			Expect(WatcherDecisionEngine.Spec.Secret).Should(Equal("test-osp-secret"))
+			Expect(WatcherDecisionEngine.Spec.MemcachedInstance).Should(Equal("memcached"))
+		})
+
+		It("should have the Status fields initialized", func() {
+			WatcherDecisionEngine := GetWatcherDecisionEngine(watcherTest.WatcherDecisionEngine)
+			Expect(WatcherDecisionEngine.Status.ObservedGeneration).To(Equal(int64(0)))
+		})
+
+		It("should have ReadyCondition false", func() {
+			th.ExpectCondition(
+				watcherTest.WatcherDecisionEngine,
+				ConditionGetterFunc(WatcherDecisionEngineConditionGetter),
+				condition.ReadyCondition,
+				corev1.ConditionFalse,
+			)
+		})
+
+		It("should have input not ready", func() {
+			th.ExpectCondition(
+				watcherTest.WatcherDecisionEngine,
+				ConditionGetterFunc(WatcherDecisionEngineConditionGetter),
+				condition.InputReadyCondition,
+				corev1.ConditionFalse,
+			)
+		})
+
+		It("should have service config input unknown", func() {
+			th.ExpectCondition(
+				watcherTest.WatcherDecisionEngine,
+				ConditionGetterFunc(WatcherDecisionEngineConditionGetter),
+				condition.ServiceConfigReadyCondition,
+				corev1.ConditionUnknown,
+			)
+		})
+
+		It("should have a finalizer", func() {
+			// the reconciler loop adds the finalizer so we have to wait for
+			// it to run
+			Eventually(func() []string {
+				return GetWatcherDecisionEngine(watcherTest.WatcherDecisionEngine).Finalizers
+			}, timeout, interval).Should(ContainElement("openstack.org/watcherdecisionengine"))
+		})
+	})
+	When("the secret is created with all the expected fields", func() {
+		BeforeEach(func() {
+			secret := th.CreateSecret(
+				watcherTest.InternalTopLevelSecretName,
+				map[string][]byte{
+					"WatcherPassword": []byte("service-password"),
+				},
+			)
+			DeferCleanup(k8sClient.Delete, ctx, secret)
+			mariadb.CreateMariaDBDatabase(watcherTest.WatcherDatabaseName.Namespace, watcherTest.WatcherDatabaseName.Name, mariadbv1.MariaDBDatabaseSpec{})
```

Comment:
this DB creation (lines 114-121) is not needed at this point, it's ok to leave it, as it will be needed for further iteartions, just fyi

Reply:
ack thanks then lets leave it here unless it creates problems to do so. as you said i'd like to get this into mergable state this week and start iterating